File: src/com/pwootage/fasterflux/CommonProxy.java
package com.pwootage.fasterflux;

import com.pwootage.fasterflux.blocks.te.TileEntityBatteryController;
import com.pwootage.fasterflux.gui.GuiBatteryController;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.world.World;
import cpw.mods.fml.common.network.IGuiHandler;

public class CommonProxy implements IGuiHandler {

	@Override
	public Object getServerGuiElement(int ID, EntityPlayer player, World world,
			int x, int y, int z) {
		return null;
	}

	@Override
	public Object getClientGuiElement(int ID, EntityPlayer player, World world,
			int x, int y, int z) {
		System.out.println("What's this? DId this work? Idk even ");
		//Because I'm too cool for gui-ids and instead infer what to open based on what I'm clicking.
		//At least for now?
		TileEntity te = world.getBlockTileEntity(x, y, z);
		if (te != null && te instanceof TileEntityBatteryController) {
			return new GuiBatteryController((TileEntityBatteryController) te);
		}
		return null;
	}

}

File: src/main/scala/com/pwootage/fasterflux/CommonProxy.scala
package com.pwootage.fasterflux

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.world.World
import cpw.mods.fml.common.network.IGuiHandler
import com.pwootage.fasterflux.blocks.te.TileEntityBatteryController
import com.pwootage.fasterflux.gui.GuiBatteryController

class CommonProxy extends IGuiHandler {

  override def getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = {
    null
  }

  override def getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): AnyRef = {
    println("What's this? DId this work? Idk even ")
    val te = world.getBlockTileEntity(x, y, z)
    if (te != null && te.isInstanceOf[TileEntityBatteryController]) {
      return new GuiBatteryController(te.asInstanceOf[TileEntityBatteryController])
    }
    null
  }
}